Source/WebCore/dom/Document.h

```diff
@@ -2620,6 +2626,7 @@ class Document
 
     mutable std::unique_ptr<CSSParserContext> m_cachedCSSParserContext;
     mutable std::unique_ptr<PermissionsPolicy> m_permissionsPolicy;
+    mutable std::unique_ptr<Style::AnchorPositionEvaluator> m_anchorPositionEvaluator;
```

Comment:
This could go to `Style::Scope` where m_queryContainerStates are. 

It is slightly awkward since they are only used for the top-level scope, not shadow tree scopes. Might still be good for consistency and keeping Document from getting even more unwieldy (Style::Scope should be split into top-level scope class and shadow level scope class at some point).